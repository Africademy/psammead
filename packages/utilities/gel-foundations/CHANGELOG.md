# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
