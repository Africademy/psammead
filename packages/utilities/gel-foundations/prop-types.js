module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
