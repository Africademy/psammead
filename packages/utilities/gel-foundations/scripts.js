/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
