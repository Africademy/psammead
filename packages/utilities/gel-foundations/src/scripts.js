/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
