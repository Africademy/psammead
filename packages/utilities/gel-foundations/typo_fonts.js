module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
