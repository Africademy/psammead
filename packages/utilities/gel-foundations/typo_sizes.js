module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
