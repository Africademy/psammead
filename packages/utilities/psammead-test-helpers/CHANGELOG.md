# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
