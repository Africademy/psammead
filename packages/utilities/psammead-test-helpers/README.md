# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactjs.org/docs/how-to-contribute.html#your-first-pull-request)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install @bbc/psammead-test-helpers --save
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
