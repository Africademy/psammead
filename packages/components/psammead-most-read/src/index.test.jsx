import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { arabic, latin } from '@bbc/gel-foundations/scripts';
import MostRead from '.';
import loadItems from './testHelpers/fixtureData';

describe('MostRead', () => {
  shouldMatchSnapshot(
    'should render with ltr most read with correct dir',
    <MostRead
      items={loadItems(10, 'LTR')}
      service="news"
      script={latin}
      dir="ltr"
      header="Most Read"
    />,
  );
  shouldMatchSnapshot(
    'should render with rtl most read with correct dir',
    <MostRead
      items={loadItems(10, 'RTL')}
      service="arabic"
      script={arabic}
      dir="rtl"
      header="الأكثر قراءة"
    />,
  );
});
