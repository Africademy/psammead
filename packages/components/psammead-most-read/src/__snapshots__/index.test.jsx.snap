// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostRead should render with ltr most read with correct dir 1`] = `
.c3 {
  position: relative;
}

.c9 {
  color: #222222;
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:hover,
.c9:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8 {
  padding-bottom: 1.5rem;
  padding-left: 1rem;
}

.c6 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-style: normal;
  color: #B80000;
  font-size: 2rem;
  line-height: 2.25rem;
  margin: 0;
  padding: 0;
  display: inline-block;
  width: 3rem;
}

.c1 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #3F3F42;
}

@media (max-width:14.9375rem) {
  .c3 {
    margin: 0 -0.25rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c3 {
    margin: 0 -0.25rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c3 {
    margin: 0 -0.25rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c3 {
    margin: 0 -0.5rem;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c3 {
    margin: 0 -0.5rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    margin: 0 -0.5rem;
  }
}

@supports (display:grid) {
  .c3 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c4 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c4 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c4 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c4 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c4 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c4 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c4 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c5 {
    width: calc(11.11111111111111%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c5 {
    width: calc(11.11111111111111%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c5 {
    width: calc(22.22222222222222%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c5 {
    width: calc(22.22222222222222%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c5 {
    width: calc(16.666666666666668%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c5 {
    width: calc(16.666666666666668%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c5 {
    display: block;
  }
}

@media (max-width:14.9375rem) {
  .c7 {
    width: calc(88.88888888888889%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c7 {
    width: calc(88.88888888888889%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c7 {
    width: calc(77.77777777777777%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c7 {
    width: calc(77.77777777777777%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c7 {
    width: calc(83.33333333333333%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c7 {
    width: calc(83.33333333333333%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c7 {
    display: block;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    padding: 0 0.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@supports (display:grid) {
  .c2 {
    padding: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:80rem) {
  .c8 {
    padding-top: 0.5rem;
    padding-left: 2rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

<div>
  <section
    aria-labelledby="most-read-title"
    class=""
    role="region"
  >
    <h2
      class="c0"
      dir="ltr"
      id="most-read-title"
    >
      Most Read
    </h2>
  </section>
  <ol
    class="c1"
    role="list"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        dir="ltr"
      >
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              1
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              2
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              3
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              4
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              5
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              6
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              7
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              8
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              9
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="ltr"
        >
          <div
            class="c5"
            dir="ltr"
          >
            <span
              class="c6"
            >
              10
            </span>
          </div>
          <div
            class="c7"
            dir="ltr"
          >
            <div
              class="c8"
              dir="ltr"
            >
              <a
                class="c9"
                href="https://www.bbc.co.uk/news/business-43328806"
              >
                John Lewis staff bonus cut again as profits fall
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </ol>
</div>
`;

exports[`MostRead should render with rtl most read with correct dir 1`] = `
.c3 {
  position: relative;
}

.c9 {
  color: #222222;
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:hover,
.c9:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8 {
  padding-bottom: 1.5rem;
  padding-right: 1rem;
}

.c6 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #B80000;
  font-size: 2rem;
  line-height: 2.25rem;
  margin: 0;
  padding: 0;
  display: inline-block;
  width: 3rem;
}

.c1 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c0 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 1.5rem;
  line-height: 2.125rem;
  color: #3F3F42;
}

@media (max-width:14.9375rem) {
  .c3 {
    margin: 0 -0.25rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c3 {
    margin: 0 -0.25rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c3 {
    margin: 0 -0.25rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c3 {
    margin: 0 -0.5rem;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c3 {
    margin: 0 -0.5rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    margin: 0 -0.5rem;
  }
}

@supports (display:grid) {
  .c3 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c4 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c4 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c4 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c4 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c4 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c4 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c4 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c5 {
    width: calc(11.11111111111111%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c5 {
    width: calc(11.11111111111111%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c5 {
    width: calc(22.22222222222222%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c5 {
    width: calc(22.22222222222222%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c5 {
    width: calc(16.666666666666668%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c5 {
    width: calc(16.666666666666668%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c5 {
    display: block;
  }
}

@media (max-width:14.9375rem) {
  .c7 {
    width: calc(88.88888888888889%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c7 {
    width: calc(88.88888888888889%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c7 {
    width: calc(77.77777777777777%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c7 {
    width: calc(77.77777777777777%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c7 {
    width: calc(83.33333333333333%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c7 {
    width: calc(83.33333333333333%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c7 {
    display: block;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    padding: 0 0.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c2 {
    padding: 0 1rem;
  }
}

@supports (display:grid) {
  .c2 {
    padding: 0;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c9 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c9 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@media (min-width:80rem) {
  .c8 {
    padding-top: 0.5rem;
    padding-right: 2rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 2rem;
    line-height: 2.375rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 2.25rem;
    line-height: 3.125rem;
  }
}

<div>
  <section
    aria-labelledby="most-read-title"
    class=""
    role="region"
  >
    <h2
      class="c0"
      dir="rtl"
      id="most-read-title"
    >
      الأكثر قراءة
    </h2>
  </section>
  <ol
    class="c1"
    role="list"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        dir="rtl"
      >
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۱
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۲
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۳
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۴
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۵
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۶
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۷
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۸
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۹
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
        <div
          as="li"
          class="c4"
          dir="rtl"
        >
          <div
            class="c5"
            dir="rtl"
          >
            <span
              class="c6"
            >
              ۱۰
            </span>
          </div>
          <div
            class="c7"
            dir="rtl"
          >
            <div
              class="c8"
              dir="rtl"
            >
              <a
                class="c9"
                href="https://www.bbc.com/persian/iran-46965704"
              >
                ایران از لغو 'رزمایش قطع اینترنت' خبر داد
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </ol>
</div>
`;
